signer/threshold_ed25519_signature.go

```diff
@@ -0,0 +1,379 @@
+package signer
+
+import (
+	"bytes"
+	"crypto"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/sha256"
+	"errors"
+	"fmt"
+	"sync"
+
+	tmJson "github.com/tendermint/tendermint/libs/json"
+	"gitlab.com/polychainlabs/edwards25519"
+	tsed25519 "gitlab.com/polychainlabs/threshold-ed25519/pkg"
+)
+
+// Interface for which is used by local  Signer
+type ThresholdEd25519Signature interface {
+	DealShares(req CosignerGetEphemeralSecretPartRequest) (HrsMetadata, error)
+
+	GetEphemeralSecretPart(req CosignerGetEphemeralSecretPartRequest) (CosignerEphemeralSecretPart, error)
+
+	SetEphemeralSecretPart(req CosignerSetEphemeralSecretPartRequest) error
+
+	Sign(req CosignerSignRequest) (CosignerSignResponse, error)
+}
+
+// LocalSoftsignThresholdEd25519Signature
+type LocalSoftSignThresholdEd25519Signature struct {
+	//UnimplementedThresholdEd25519Signature // embedding unimplemented ThresholdEd25519Signature
+	PubKeyBytes []byte
+	Key         CosignerKey
+	RsaKey      rsa.PrivateKey
+	Total       uint8
+	Threshold   uint8
+
+	// stores the last sign state for a share we have fully signed
+	// incremented whenever we are asked to sign a share
+	LastSignState *SignState
```

Comment:
`LastSignState`, `LastSignStateMutex`, and `Peers` should move back out to the LocalCosigner. The `ThresholdEd25519Signature` interface methods can be updated to pass in the necessary parameters.